services/venice-server/src/main/java/com/linkedin/venice/listener/StatsHandler.java

```diff
@@ -245,6 +250,62 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) {
     }
   }
 
+  @Override
+  public void grpcRead(GrpcHandlerContext ctx, GrpcHandlerPipeline pipeline) {
+    if (ctx.hasError()) {
+      pipeline.processRequest(ctx);
+      return;
+    }
+
+    if (ctx.getGrpcStatsContext() == null) {
+      // new request
+      GrpcStatsContext statsContext = new GrpcStatsContext(singleGetStats, multiGetStats, computeStats);
+
+      ctx.setGrpcStatsContext(statsContext);
+      pipeline.processRequest(ctx);
+    } else {
+      // existing request
+      pipeline.processRequest(ctx);
```

Comment:
If this is the same statement at the end of both branches of the if, then we can move it out of the if, and always execute it after.